base/_utils.py

```diff
@@ -122,7 +122,172 @@ def __cmp__(self, other):
+
+    class code_2x(object):
+        def __new__(cls, object):
+            return object.func_code
+        @classmethod
+        def name(cls, object):
+            return object.co_name
+        @classmethod
+        def flags(cls, object):
+            return object.co_flags
+        @classmethod
+        def argcount(cls, object):
+            return object.co_argcount
+        @classmethod
+        def varnames(cls, object):
+            return object.co_varnames
+
+        cons = collections.namedtuple('code_t', ['co_argcount', 'co_nlocals', 'co_stacksize', 'co_flags', 'co_code', 'co_consts', 'co_names', 'co_varnames', 'co_filename', 'co_name', 'co_firstlineno', 'co_lnotab', 'co_freevars', 'co_cellvars'])
+        @classmethod
+        def unpack(cls, object):
+            return cls.cons(*(getattr(object, item) for item in cls.cons._fields))
+        @classmethod
+        def unpack_extra(cls, object):
+            return ()
+        @classmethod
+        def new(cls, attributes, extra=()):
+            argcount, nlocals, stacksize, flags, code, consts, names, varnames, filename, name, firstlineno, lnotab, freevars, cellvars = attributes
+            return types.CodeType(*attributes)
+
+    class code_3x(code_2x):
+        def __new__(cls, object):
+            return object.__code__
+        @classmethod
+        def unpack_extra(cls, object):
+            return object.co_posonlyargcount, object.co_kwonlyargcount
+        @classmethod
+        def new(cls, attributes, extra=(0, 0)):
+            argcount, nlocals, stacksize, flags, code, consts, names, varnames,filename, name, firstlineno, lnotab, freevars, cellvars = attributes
+            posonlyargcount, kwonlyargcount = extra
+            return types.CodeType(argcount, posonlyargcount, kwonlyargcount, nlocals, stacksize, flags, code, consts, names, varnames, filename, name, firstlineno, lnotab, freevars, cellvars)
+
+    code = code_2x if sys.version_info.major < 3 else code_3x
+
+    class method_2x(object):
+        @classmethod
+        def new(cls, function, instance, type):
+            return types.MethodType(function, instance, type)
+
+        @classmethod
+        def self(cls, object):
+            return object.im_self
+
+        @classmethod
+        def type(cls, object):
+            return object.im_class
+
+        @classmethod
+        def function(cls, object):
+            return object.im_func
+
+    class method_3x(object):
+        @classmethod
+        def new(cls, function, instance, type=None):
+            return types.MethodType(function, instance)
+
+        @classmethod
+        def self(cls, object):
+            return object.__self__
+
+        @classmethod
+        def type(cls, object):
+            return object.__self__.__class__
+
+        @classmethod
+        def function(cls, object):
+            return object.__func__
+
+    method = method_2x if sys.version_info.major < 3 else method_3x
+
 ### decorators
+class heap_package(object):
```

Comment:
I think this should be renamed to something like `priority_tuple` or something...